geps/gep-1897.md

```diff
@@ -125,7 +141,326 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
 
 ## API
 
-Details deferred until we reach consensus on what we want to do, and why we want to do this.
+To allow the gateway client to know how to connect to the backend pod, when the backend pod has its own
+certificate, we implement a metaresource named `TLSConnectionPolicy`, that is already mentioned as an example in
+[GEP-713: Metaresources and PolicyAttachment](https://gateway-api.sigs.k8s.io/geps/gep-713/), and as a hypothetical in the
+[Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment)
+documentation.
+
+In this document, because naming is hard, we chose to retain the
+name TLSConnectionPolicy to advertise alignment with a previously discussed naming choice, but a new name may be
+substituted without blocking acceptance of the content of the API change.
+
+The selection of the applicable Gateway API persona is important in the design of TLSConnectionPolicy, because it provides
+a way to explicitly describe the _expectations_ of the connection to the application.  TLSConnectionPolicy is configured
+by the application developer Gateway API persona to signal what the application developer _expects_ in connections to
+the application, from a TLS perspective.  Only the application developer can know what the application expects, so it is
+important that this configuration be managed by that persona.
+
+During the course of discussion of this proposal, we did consider allowing the cluster operator persona to have some access
+to Gateway cert validation, but as mentioned, TLSConnectionPolicy is used primarily to signal what the application
+developer expects in the connection.  Granting this expectation to any other role would blur the lines between role
+responsibilities, which violates the role-oriented design principle of Gateway API. As mentioned in Non-goal #7,
+providing a mechanism for the cluster operator gateway role to override gateway to backend TLS settings is not covered
+by this proposal, but should be addressed in a future update.  One idea is to use two types: ApplicationTLSConnectionPolicy,
+and GatewayTLSConnectionPolicy, where the application developer is responsible for the former, the cluster operator is
+responsible for the latter, and the cluster operator may configure whether certain settings may be overridden by
+application developers.
+
+Using the viewpoint of the application running inside the Pods, the TLSConnectionPolicy should contain configuration
+for these use cases of the Gateway as a TLS Client:
+
+- An explicit signal that TLS should be used by this connection.
+- Which Certificate Authority (CA) to use, whether the system bundled CA, an internal CA, or self-signed.
+- TLS version to expect.  This allows implementations to introduce features to disallow or force certain TLS versions.
```

Comment:
Isn't the TLS version negotiation sufficient ? Backend can choose min TLS version it accepts, if gateway doesn't support it - nothing will help.

Reply:
It doesn't seem like an overcomplicated config field.  I choose to keep it in unless there are other objections.